config/routes.rb

```diff
@@ -23,5 +23,10 @@
   get '/conditions/edit', to: 'conditions#edit'
   post '/conditions/create', to: 'conditions#create'
   get '/medicine_notebook/search', to: 'medicine_notebook_records#search'
+  get '/medicine_notebook/new/:id', to: 'medicine_notebook_records#new', as: :medicine_notebook_new
+  post '/medicine_notebook/new/:id', to: 'medicine_notebook_records#create'
```

Comment:
作成時は `:id` がないはずなので、こちらのほうがよさそうです。
```suggestion
  post '/medicine_notebook', to: 'medicine_notebook_records#create'
```